File: src/main/java/com/paragon/client/ui/alt/AltEntry.java
package com.paragon.client.ui.alt;

import com.paragon.api.util.Wrapper;
import com.paragon.api.util.render.RenderUtil;
import com.paragon.api.util.render.ITextRenderer;
import com.paragon.client.managers.alt.Alt;

public final class AltEntry implements ITextRenderer, Wrapper {

    private final Alt alt;
    private float offset;

    public AltEntry(Alt alt, float offset) {
        this.alt = alt;
        this.offset = offset;
    }

    public void drawAlt(int mouseX, int mouseY, int screenWidth) {
        RenderUtil.drawRect(0, offset, screenWidth, 20, AltManagerGUI.selectedAltEntry == this ? 0x95111111 : 0x95000000);
        renderCenteredString(alt.getEmail(), screenWidth / 2f, offset + 10, -1, true);
    }

    public void clicked(int mouseX, int mouseY, int screenWidth) {
        if (isHovered(0, offset, screenWidth, 20, mouseX, mouseY)) {
            alt.login();
        }
    }

    public float getOffset() {
        return offset;
    }

    public void setOffset(float newOffset) {
        this.offset = newOffset;
    }

    public Alt getAlt() {
        return alt;
    }

}

File: src/main/kotlin/com/paragon/client/ui/alt/AltEntry.kt
package com.paragon.client.ui.alt

import com.paragon.api.util.Wrapper
import com.paragon.api.util.render.RenderUtil.drawRect
import com.paragon.api.util.render.font.FontUtil.renderCenteredString
import com.paragon.client.managers.alt.Alt

class AltEntry(val alt: Alt, var offset: Float) : Wrapper {

    fun drawAlt(mouseX: Int, mouseY: Int, screenWidth: Int) {
        drawRect(0f, offset, screenWidth.toFloat(), 20f, if (AltManagerGUI.selectedAltEntry === this) -0x6aeeeeef else -0x6b000000)
        renderCenteredString(alt.email, screenWidth / 2f, offset + 10, -1, true)
    }

    fun clicked(mouseX: Int, mouseY: Int, screenWidth: Int) {
        if (isHovered(0f, offset, screenWidth.toFloat(), 20f, mouseX, mouseY)) {
            alt.login()
        }
    }

}